bdd(nfs-retain-pvc): verify volume-create events for PVC with Retain policy creation when volume-events-exporter is disabled

mynktl opened the pull request:
Changes:
This commit adds a sanity test to verify volume-create events for the PVC, having the reclaim Policy set
to Retained and volume-event-exporter is disabled. Since the volume-event-exporter is disabled,
the event-listener won't receive the volume-create events. Once the volume-event-exporter is enabled,
It will start sending the event for the PVCs created when the exporter was disabled or not running.

Signed-off-by: mayank <mayank.patel@mayadata.io>

mittachaitu commented on tests/nfs_create_retain_pv_disable_exporter_test.go:
```diff
@@ -0,0 +1,311 @@
+				backendPVCObj, err := Client.getPVC(OpenEBSNamespace, backendPVCName)
+				Expect(err).To(BeNil(), "while fetching backend pvc %s/%s", OpenEBSNamespace, backendPVCName)
+
+				_, isCreateEventReceived = backendPVCObj.Annotations[nfs.VolumeDeleteNFSPVKey]
```
We need to check for create event key, since we are checking for create key
```suggestion
				_, isCreateEventReceived = backendPVCObj.Annotations[nfs.VolumeCreateNFSPVKey]
```

mittachaitu reviewed (commented):
One minor comment... Rest looks good

Commit:
bdd(volume-event-controller): test to verify events for NFS PVC creation when volume-event-controller is disabled

This commit adds a sanity test to verify volume-create events for the PVC, having the reclaim Policy set
to Retain, and volume-event-exporter is disabled. Since the volume-event-exporter is disabled,
the event-listener won't receive the volume-create events. Once the volume-event-exporter is enabled or
starts running, It will continue sending the event for the PVCs created when the exporter was disabled or not running.

Signed-off-by: mayank <mayank.patel@mayadata.io>

Commit:
updating go mod files

Signed-off-by: mayank <mayank.patel@mayadata.io>

mittachaitu commented on tests/nfs_create_retain_pv_disable_exporter_test.go:
```diff
@@ -0,0 +1,329 @@
+		})
+	})
+
+	When("PVC with storageclass openebs-rwx is created", func() {
```
```suggestion
	When("PVC with storageclass "+scName +" is created", func() {
```

mittachaitu commented on tests/nfs_create_retain_pv_disable_exporter_test.go:
```diff
@@ -0,0 +1,329 @@
+
+			err = Client.deletePVC(OpenEBSNamespace, backendPVCName)
+			Expect(err).To(BeNil(), "while deleting backend pvc %s/%s", OpenEBSNamespace, backendPVCName)
+
```
We also need to delete the NFS Service resource (or) let's leave it to garbage collector

mynktl replied:
we can add that since it is only missing. we can leave it to garbage collector also but may conflict if the future test is having the same pvc name. I'll add deletion for service

mittachaitu commented on tests/nfs_sanity_test.go:
```diff
@@ -29,7 +29,7 @@ import (
 )
 
-var _ = Describe("TEST NFS PVC CREATE & DELTE EVENTS", func() {
+var _ = Describe("TEST NFS PVC CREATE & DELETE EVENTS", func() {
```
:+1:

mittachaitu reviewed (commented):
Looks good... just found that we are not deleting the Service resource created for the test

Commit:
Addressing review comments

Signed-off-by: mayank <mayank.patel@mayadata.io>

mittachaitu reviewed (approved):
changes are good

Commit:
Merge branch 'develop' of github.com:mayadata-io/volume-events-exporter into bdd_nfs_disable_exporter_create_ops

Signed-off-by: mayank <mayank.patel@mayadata.io>
